mavros/src/plugins/setpoint_attitude.cpp

```diff
@@ -215,6 +237,43 @@ class SetpointAttitudePlugin : public plugin::PluginBase,
 			send_attitude_ang_velocity(req->header.stamp, ang_vel, thrust_msg->thrust);
 	}
 
+	void rpyt_cb(const mav_msgs::RollPitchYawrateThrustConstPtr msg)
+	{
+		if(ignore_rpyt_messages_){
+			ROS_FATAL("Recieved roll_pitch_yaw_thrust_rate message, but ignore_rpyt_messages_ is true: "
+	 	        "the most likely cause of this is a failure to specify the thrust_scaling_factor,	"
+			"yaw_rate_scaling_factor or system_mass_kg parameters");
+		     	return;
+		}
+
+		// Set mask to ignore everything but thrust and attitude setpoint
+		uint8_t ignore_all_except_q_and_thrust = (7 << 0);
+
+		geometry_msgs::Quaternion orientation = 
+			tf::createQuaternionMsgFromRollPitchYaw( msg->roll, msg->pitch, 0);
+		// Transforms from thrust acceleration to a thrust force (scaling adjusts to the UAV
```

Comment:
This is incorrect. The thrust in the message, according to the definition in mav_msgs, is in Newtons (i.e a force) already. Can you also describe here how you compute the scaling factor in order to get a thrust value normalized from 0 to 1?

Reply:
I'm trying to figure out the best way to apply that scaling, from newtons [0,1]. On the UAV, 1 is the maximum thrust it can generate right? If so, a previously calibration needs to be set, and thrust_scaling set.  

The scaling can be done by putting the drone on a scale. Let's say the drone weights 2kg [thrust = 0] :
F=ma, so F=2*9.81= 19,62

When it lifts one kg [say at thrust.z = 0.1], then we know that it applied 9.81N, At maximum, it can do 98.1N of Force. 

So, 98.1N ----> thrust. = 1.0
         req N  ----->  x , x = req*1.0/98.1 , and so the scaling factor is  1/98.1 =  0.01019 . 

So, finally, we can set thrust = std::max(0.0, msg->thrust.z * thrust_scaling) , on which thrust_scaling is given by the user.

Reply:
Okay, all this makes sense. We keep the thrust scaling in the plugin.

> this implies that the thrust is linearly increasing

I think this is a fair assumption. We can handle linearisation on the autopilot side.

Reply:
Thanks @mhkabir , I'll update this later in the day.